internal/moreos/moreos_func-e_test.go

```diff
@@ -45,68 +46,106 @@ var (
-	requireFindProcessError(t, cmd.Process, process.ErrorProcessNotRunning)
-	requireFindProcessError(t, fakeEnvoyProcess, process.ErrorProcessNotRunning)
+	tests := []testCase{
+		{
+			name:           "interrupt",
+			signal:         Interrupt,
+			waitForExiting: true,
+			skip:           runtime.GOOS == OSWindows,
+		},
+		{
+			name:           "SIGTERM",
+			signal:         func(proc *os.Process) error { return proc.Signal(syscall.SIGTERM) },
+			waitForExiting: true,
+			// On Windows, os.Process.Signal is not implemented; it will return an error instead of sending
+			// a signal.
+			skip: runtime.GOOS == OSWindows,
+		},
+		{
+			name: "kill",
+			// On Linux, we propagate SIGKILL to the child process as the configured SysProcAttr.Pdeathsig
+			// in proc_linux.go.
+			signal:         func(proc *os.Process) error { return proc.Kill() },
+			waitForExiting: false, // since the process is killed, it is immediately exit.
+			// This works only for Linux, sending kill -9 on Darwin will not kill the process, we need to
+			// kill via pgid or kill the child first.
+			skip: runtime.GOOS != OSLinux,
+		},
+	}
 
-	// Ensure both processes are killed.
-	require.NoError(t, EnsureProcessDone(cmd.Process))
-	require.NoError(t, EnsureProcessDone(fakeEnvoyProcess))
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.skip {
+				t.Skip()
+			}
+
+			tempDir := t.TempDir()
+
+			// Build a fake envoy and pass the ENV hint so that fake func-e uses it
+			fakeEnvoy := filepath.Join(tempDir, "envoy"+Exe)
+			fakebinary.RequireFakeEnvoy(t, fakeEnvoy)
+			t.Setenv("ENVOY_PATH", fakeEnvoy)
+
+			fakeFuncE := filepath.Join(tempDir, "func-e"+Exe)
+			requireFakeFuncE(t, fakeFuncE)
+
+			stdout := new(bytes.Buffer)
+
+			// With an arg so fakeFuncE runs fakeEnvoy as its child and doesn't exit.
+			arg := string(version.LastKnownEnvoy)
+			cmd := exec.Command(fakeFuncE, "run", arg, "-c")
+			cmd.Stdout = stdout
+
+			stderr, err := cmd.StderrPipe()
+			require.NoError(t, err)
+			stderrScanner := bufio.NewScanner(stderr)
+
+			require.NoError(t, cmd.Start())
+
+			// Block until we reach an expected line or timeout.
+			requireScannedWaitFor(t, stderrScanner, "starting main dispatch loop")
+			require.Equal(t, Sprintf("starting: %s %s -c\n", fakeEnvoy, arg), stdout.String())
+
+			fakeFuncEProcess, err := process.NewProcess(int32(cmd.Process.Pid))
+			require.NoError(t, err)
+
+			// Get all fake func-e children processes.
+			children, err := fakeFuncEProcess.Children()
+			require.NoError(t, err)
+			require.Equal(t, len(children), 1) // Should have only one child process i.e. the fake envoy process.
+			fakeEnvoyProcess := &os.Process{Pid: int(children[0].Pid)}
+
+			tc.signal(cmd.Process)
+			if tc.waitForExiting {
+				requireScannedWaitFor(t, stderrScanner, "exiting")
+			}
+
+			// Wait for the process to die; this could error due to the kill signal.
+			err = cmd.Wait()
```

Comment:
On windows, this `cmd.Wait` will exit when the test runner is finished. You can kill the child alone (cmd) here.

Reply:
Resolved. That was my fault for not setting the creation flag.